.github/actions/bootstrap/action.yml

```diff
@@ -0,0 +1,28 @@
+name: bootstrap
+description: bootstrap rmf-web
+inputs:
+  package:
+    description: string, target package to bootstrap
+    required: true
+  skip-build:
+    description: boolean, if true, skip the build step
+    required: false
+    default: false
+runs:
+  using: composite
+  steps:
+    - uses: pnpm/action-setup@v2.2.2
+      with:
+        version: 'latest'
+    - uses: actions/setup-node@v2
+      with:
+        node-version: '16'
+        cache: 'pnpm'
+    - uses: Gr1N/setup-poetry@v7
```

Comment:
```suggestion
    - uses: Gr1N/setup-poetry@v7
    - with:
        poetry-version: 1.1.15
```

Reply:
sry, I meant pinning the version of the offending package. But after reading up more on the problem, I'm not sure how to proceed, if the offending package is from the system packages (which it probably is because one of the comments says that PyPI doesn't allow non PEP-440 versionings), then there is nothing we can do.

Since this behavior is pretty much decided by the poetry community, best we can do is switch to another package manager. Maybe go back to pipenv and live with it's problems.